File: codes/medium/38 Count and Say/oldSolution.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
	
	if (n == 1) {
		return "1";
	} else {
		let str = countAndSay(n-1);
		let result = "";
		for(let i = 0; i < str.length; i++) {
			if(i > 0 && str[i] === str[i - 1]) {
				continue;
			}
			
			let counter = i;
			let num = 1;
			
			
			while(str[counter] === str[counter + 1]) {
				counter++;
				num++;
			}
			
			result += num;
			result += str[counter];
		}
		
		return result;
	}
};    

File: codes/medium/38 Count and Say/oldSolution.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n == 1) {
            return "1";
        } else {
            string str = countAndSay(n - 1);
            string result = "";

            for (int i = 0; i < str.size(); ++i) {
                if(i > 0 && str[i] == str[i - 1]) {
                    continue;
                }

                int counter = i;
                int occurences = 1;

                while(str[counter] == str[counter + 1]) {
                    counter++;
                    occurences++;
                }

                

                result += std::to_string(occurences);
                result += str[counter];
            }

            return result;
        }
    }
};